src/app/artist-details/artist-details.component.html

```diff
@@ -0,0 +1,60 @@
+<div class="card2" *ngIf="artist">
+  <div class="card-header2">
+    <div>
+      <h1>
+        <span label label-default style="align-content: center;">ARTIST PROFILE</span>
+      </h1>
+    </div>
+  </div>
+
+  <div class="card-body">
+    <div class="block-21 mb-4 d-flex" *ngIf="artist">
+      <a class="blog-img mr-4"></a>
+      <img class="center-block img-responsive" [style.width.px]="200" [style.margin.px]="2"
+        [src]="artist.picture_medium" [title]="artist.name" />
+      <div class="text">
+        <h3 class="heading">
+          <a>{{ artist.name }}</a>
+        </h3>
+        <div class="meta">
+          <div>
+            <a><span>Number of Fans:</span>
+              {{ artist.nb_fan }}
+            </a>
+          </div>
+        </div>
+      </div>
+    </div>
+    <span label label-default style="align-content: center;">ARTIST TOP TRACKS</span>
+    <div class="row">
+      <ul class="list-group" *ngFor="let tracks of tracks">
+        <li class="list-group-item">
+          {{ tracks[tracks] }} {{ tracks.title }} , {{ tracks.duration }}
+        </li>
+      </ul>
+    </div>
+  </div>
+
+  <div>
+    <h1>
+      <span label label-default style="align-content: center;">ALBUMS</span>
+    </h1>
+  </div>
+  <div class="row">
+    <div class="column" *ngFor="let album of album">
```

Comment:
The variable naming here is strange and may lead to undesired behavior.

Consider updating the directive to `*ngFor="let album of albums"`

I think you've misunderstood how `*ngFor` works, please take a look at [this](https://angular.io/guide/displaying-data) Angular example.

Reply:
@strizzwald fixed